core/src/main/java/com/scalar/db/storage/dynamo/QueryScanner.java

```diff
@@ -0,0 +1,100 @@
+package com.scalar.db.storage.dynamo;
+
+import com.scalar.db.api.Result;
+import com.scalar.db.api.Scanner;
+import com.scalar.db.storage.common.ScannerIterator;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.QueryRequest;
+import software.amazon.awssdk.services.dynamodb.model.QueryResponse;
+
+public class QueryScanner implements Scanner {
```

Comment:
I think this `QueryScanner` is the main change of this PR. This class handles the `lastEvaluatedKey`.